### PR TITLE
qa/tasks/rook: test reapplication of drive groups stored in mgr

### PR DESCRIPTION
**Pull request by josephsawaya**
This commit adds testing for the drive_group_loop in the Rook orchestrator
that reapplies drive groups that were applied previously.

This test removes an OSD, zaps the underlying device then waits for the OSD
to be re-created by the drive_group_loop.

Signed-off-by: Joseph Sawaya <jsawaya@redhat.com>

Depends on  #43138

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/suites/orch/rook/smoke/3-final.yaml [hunk 1]
```diff
@@ -6,3 +6,34 @@ tasks:
...
+    commands:
+      - |
+        toolbox() {
+            kubectl -n rook-ceph exec -it deploy/rook-ceph-tools --  "$@"
+        }
+        toolbox ceph orch osd rm 0 --force
+        removed_pv=""
+        while [ "$removed_pv" = "" ]
+        do
+            removed_pv=`kubectl get pv | grep Released | cut -f1 -d " "`
```
Comment (jeffvance) [inline qa/suites/orch/rook/smoke/3-final.yaml:19]: Likely an ignorant question, but how do you know the PV is for ceph?

File: qa/suites/orch/rook/smoke/3-final.yaml [hunk 2]
```diff
@@ -6,3 +6,34 @@ tasks:
...
+            sleep 3s
```
Comment (jeffvance) [inline qa/suites/orch/rook/smoke/3-final.yaml:20]: After some number of iterations should this loop stop?

**Comment by josephsawaya** [comment]
jenkins test make check

**Comment by josephsawaya** [comment]
jenkins test make check

**Comment by liewegas** [comment]
```
[root@rook-ceph-tools-744c55f865-9brwh /]# ceph orch osd rm 0 --force
Error cleaning environment after removing OSDs from Ceph cluster CRD: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Audit-Id': '20cad418-a263-4309-9313-ebb7ac722d54', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Kubernetes-Pf-Flowschema-Uid': '1374b1c1-b92f-403d-b423-53c7829d5151', 'X-Kubernetes-Pf-Prioritylevel-Uid': '7f1ccd3c-4a50-4ac5-a295-b46226ab8f1e', 'Date': 'Mon, 15 Nov 2021 20:34:44 GMT', 'Content-Length': '389'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"deployments.apps \"rook-ceph-osd-0\" is forbidden: User \"system:serviceaccount:rook-ceph:rook-ceph-mgr\" cannot patch resource \"deployments/scale\" in API group \"apps\" in the namespace \"rook-ceph\"","reason":"Forbidden","details":{"name":"rook-ceph-osd-0","group":"apps","kind":"deployments"},"code":403}
```

**Comment by liewegas** [comment]
```
status:
  message: No recycler plugin found for the volume!
  phase: Failed
```

**Comment by josephsawaya** [comment]
jenkins test make check

**Comment by josephsawaya** [comment]
https://pulpito.ceph.com/jsawaya-2021-12-14_15:44:05-orch:rook-wip-joseph-testing-distro-basic-smithi/

**Comment by josephsawaya** [comment]
jenkins retest this please

File: qa/tasks/rook.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 
 log = logging.getLogger(__name__)
 
+path_to_examples = 'rook/deploy/examples/'
```
Comment (sebastian-philipp) [inline qa/tasks/rook.py:25]: can we make this stateless? 

```python
def path_to_examples(root_path: str) -> str:
    for p in ['rook/deploy/examples/', ''rook/cluster/examples/kubernetes/ceph/']
        if os.path.isdir(os.path.join(root_path, p)):
            return p
    assert False. 'Path to examples not found'
```

**Comment by sebastian-philipp** [comment]
could you please add `Fixes: https://tracker.ceph.com/issues/53501` to the commit message?

**Review by sebastian-philipp — APPROVED** [review]

**Comment by josephsawaya** [comment]
https://pulpito.ceph.com/jsawaya-2021-12-17_16:59:06-orch:rook-wip-joseph-testing-distro-basic-smithi/

**Comment by ljflores** [comment]
@josephsawaya this is still happening in pacific. I think this needs PR needs a backport?

**Comment by josephsawaya** [comment]
> @josephsawaya this is still happening in pacific. I think this needs PR needs a backport?

I'm actually working on a PR to remove the orchestrator commands from the Rook task since the orchestrator is not being maintained at the moment, will mention you in that.

**Comment by ljflores** [comment]
Thanks @josephsawaya!

**Pull request merged**